proposals/lip-Introduce_PoA_module.md

```diff
@@ -0,0 +1,607 @@
+* Call `registerValidatorKeys(address, trs.params.proofOfPossession, trs.params.generatorKey, trs.params.blsKey)`, where `address` is the 20-byte address derived from `trs.senderPublicKey`.
+The function `registerValidatorKeys` is defined in the [Validators module][lip-validators-module-register-validator-keys].
+
+#### Update Generator Key Command
+
+This command is used to update the generator key (from [the Validators module][validatorsModule]) for a specific authority. 
+The command ID of this transaction is `COMMAND_ID_UPDATE_KEY`.
+
+##### Parameters
+
+```java
+updateGeneratorKeyParamsSchema = {
+  "type": "object",
+  "properties": {
+    "generatorKey": {
+      "dataType": "bytes",
+      "fieldNumber": 1
+    }
+  },
+  "required": ["generatorKey"]
+}
+```
+
+##### Verification
+
+Let `trs` be a transaction with module ID `MODULE_ID_POA` and command ID `COMMAND_ID_UPDATE_KEY` to be executed.
+The list of verification conditions for `trs.params` is as follows:
+
+* Let `address` be the 20-byte address derived from `trs.senderPublicKey`.
+Then the validators substore must have an entry for the store key `address`.
+* `trs.params.generatorKey` must have length 32.
+
+##### Execution
+
+Let `trs` be a transaction with module ID `MODULE_ID_POA` and command ID `COMMAND_ID_UPDATE_KEY` to be executed.
+Then `trs.params` implies the following execution logic:
+
+Let `address` be the 20-byte address derived from `trs.senderPublicKey`. 
+Then, call `setValidatorGeneratorKey(address, trs.params.generatorKey)`, where `setValidatorGeneratorKey` is the function exposed by [the Validators module][validatorsModule].
+
+#### Update Authority Command
+
+The command ID for this command is `COMMAND_ID_UPDATE_AUTHORITY`.
+
+##### Parameters
+
+``` java
+updateValidatorParams = {
+   "type": "object",
+   "properties": {
+      "newValidators: {
+          "type": "array",
+          "fieldNumber": 1,    
+          "items": {
+              "type": "object",
+              "required": ["address", "weight"],
+              "properties": {
+	          "address": {
+                      "dataType": "bytes",
+                      "fieldNumber": 1
+                  }
+                  "weight": {
+                      "dataType": "bytes",
+                      "fieldNumber": 2
+                  }
+	      }
+	  }
+      },
+      "threshold": {
+         "dataType": "uint64",
+         "fieldNumber": 2
+      },
+      "validatorsUpdateNonce": {
+         "dataType": "uint32",
+         "fieldNumber": 3
+      },
+      "signature": {
+         "dataType": "bytes",
+         "fieldNumber": 4
+      },
+      "aggregationBits": {
+         "dataType": "bytes",
+         "fieldNumber": 5
+      }
+   },
+   "required": [
+      "newValidators",
+      "threshold",
+      "validatorsUpdateNonce",
+      "signature",
+      "aggregationBits"
+   ]
+}
+
```

Comment:
Also above the a spaces and tabs mixed that is why the indentation in the source seems incorrect.

Reply:
fixed